PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:10 + add:0, change matches:
-    _inherit = 'calendar.event'
-    _inherit = 'digest.digest'
-    _inherit = 'hr.department'
-    _inherit = "hr.employee"
-    _inherit = 'ir.ui.menu'
-    _inherit = 'mail.activity.plan'
-    _inherit = "res.company"
-    _inherit = 'res.users'
-    _inherit = 'utm.campaign'
-    _inherit = 'utm.source'

Total Changes: 60

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/calendar.py
+++ b/addons/hr_recruitment/models/calendar.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class CalendarEvent(models.Model):
-    _inherit = 'calendar.event'
+    _inherit = ['calendar.event']
 
     @api.model
     def default_get(self, fields):

--- a/addons/hr_recruitment/models/digest.py
+++ b/addons/hr_recruitment/models/digest.py
@@ -5,8 +5,8 @@ from odoo import fields, models, _
 from odoo.exceptions import AccessError
 
 
-class Digest(models.Model):
-    _inherit = 'digest.digest'
+class DigestDigest(models.Model):
+    _inherit = ['digest.digest']
 
     kpi_hr_recruitment_new_colleagues = fields.Boolean('New Employees')
     kpi_hr_recruitment_new_colleagues_value = fields.Integer(compute='_compute_kpi_hr_recruitment_new_colleagues_value')

--- a/addons/hr_recruitment/models/hr_applicant.py
+++ b/addons/hr_recruitment/models/hr_applicant.py
@@ -19,8 +19,7 @@ AVAILABLE_PRIORITIES = [
 ]
 
 
-class Applicant(models.Model):
-    _name = "hr.applicant"
+class HrApplicant(models.Model):
     _description = "Applicant"
     _order = "priority desc, id desc"
     _inherit = ['mail.thread.cc',
@@ -511,7 +510,7 @@ class Applicant(models.Model):
         }
 
     def _track_template(self, changes):
-        res = super(Applicant, self)._track_template(changes)
+        res = super()._track_template(changes)
         applicant = self[0]
         # When applcant is unarchived, they are put back to the default stage automatically. In this case,
         # don't post automated message related to the stage change.
@@ -533,7 +532,7 @@ class Applicant(models.Model):
         record = self[0]
         if 'stage_id' in init_values and record.stage_id:
             return self.env.ref('hr_recruitment.mt_applicant_stage_changed')
-        return super(Applicant, self)._track_subtype(init_values)
+        return super()._track_subtype(init_values)
 
     def _notify_get_reply_to(self, default=None):
         """ Override to set alias of applicants to their job definition if any. """
@@ -541,7 +540,7 @@ class Applicant(models.Model):
         res = {app.id: aliases.get(app.job_id.id) for app in self}
         leftover = self.filtered(lambda rec: not rec.job_id)
         if leftover:
-            res.update(super(Applicant, leftover)._notify_get_reply_to(default=default))
+            res.update(super(HrApplicant, leftover)._notify_get_reply_to(default=default))
         return res
 
     def _message_get_suggested_recipients(self):
@@ -623,7 +622,7 @@ class Applicant(models.Model):
                 self.search([
                     ('partner_id', '=', False), email_domain, ('stage_id.fold', '=', False)
                 ]).write({'partner_id': new_partner[0].id})
-        return super(Applicant, self)._message_post_after_hook(message, msg_vals)
+        return super()._message_post_after_hook(message, msg_vals)
 
     def create_employee_from_applicant(self):
         self.ensure_one()
@@ -667,7 +666,7 @@ class Applicant(models.Model):
 
     def toggle_active(self):
         self = self.with_context(just_unarchived=True)
-        res = super(Applicant, self).toggle_active()
+        res = super().toggle_active()
         active_applicants = self.filtered(lambda applicant: applicant.active)
         if active_applicants:
             active_applicants.reset_applicant()

--- a/addons/hr_recruitment/models/hr_applicant_category.py
+++ b/addons/hr_recruitment/models/hr_applicant_category.py
@@ -6,8 +6,7 @@ from random import randint
 from odoo import fields, models
 
 
-class ApplicantCategory(models.Model):
-    _name = "hr.applicant.category"
+class HrApplicantCategory(models.Model):
     _description = "Category of applicant"
 
     def _get_default_color(self):

--- a/addons/hr_recruitment/models/hr_applicant_refuse_reason.py
+++ b/addons/hr_recruitment/models/hr_applicant_refuse_reason.py
@@ -4,8 +4,7 @@
 from odoo import fields, models
 
 
-class ApplicantRefuseReason(models.Model):
-    _name = "hr.applicant.refuse.reason"
+class HrApplicantRefuseReason(models.Model):
     _description = 'Refuse Reason of Applicant'
     _order = 'sequence'
 

--- a/addons/hr_recruitment/models/hr_candidate.py
+++ b/addons/hr_recruitment/models/hr_candidate.py
@@ -10,7 +10,6 @@ from odoo.osv import expression
 
 
 class HrCandidate(models.Model):
-    _name = "hr.candidate"
     _description = "Candidate"
     _inherit = ['mail.thread.cc',
                'mail.thread.main.attachment',

--- a/addons/hr_recruitment/models/hr_department.py
+++ b/addons/hr_recruitment/models/hr_department.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class HrDepartment(models.Model):
-    _inherit = 'hr.department'
+    _inherit = ['hr.department']
 
     new_applicant_count = fields.Integer(
         compute='_compute_new_applicant_count', string='New Applicant', compute_sudo=True)

--- a/addons/hr_recruitment/models/hr_employee.py
+++ b/addons/hr_recruitment/models/hr_employee.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class HrEmployee(models.Model):
-    _inherit = "hr.employee"
+    _inherit = ["hr.employee"]
 
     # YTI Rename into candidate_ids
     candidate_id = fields.One2many('hr.candidate', 'employee_id', 'Candidate', groups="hr.group_hr_user")

--- a/addons/hr_recruitment/models/hr_job.py
+++ b/addons/hr_recruitment/models/hr_job.py
@@ -9,8 +9,7 @@ from odoo.tools import SQL
 from odoo.tools.convert import convert_file
 
 
-class Job(models.Model):
-    _name = "hr.job"
+class HrJob(models.Model):
     _inherit = ["mail.alias.mixin", "hr.job"]
     _order = "sequence, name asc"
 

--- a/addons/hr_recruitment/models/hr_job_platform.py
+++ b/addons/hr_recruitment/models/hr_job_platform.py
@@ -4,8 +4,7 @@ from odoo import api, fields, models
 from odoo.tools import email_normalize
 
 
-class JobPlatform(models.Model):
-    _name = "hr.job.platform"
+class HrJobPlatform(models.Model):
     _description = 'Job Platforms'
 
     name = fields.Char(required=True)

--- a/addons/hr_recruitment/models/hr_recruitment_degree.py
+++ b/addons/hr_recruitment/models/hr_recruitment_degree.py
@@ -4,8 +4,7 @@
 from odoo import fields, models
 
 
-class RecruitmentDegree(models.Model):
-    _name = "hr.recruitment.degree"
+class HrRecruitmentDegree(models.Model):
     _description = "Applicant Degree"
 
     name = fields.Char("Degree Name", required=True, translate=True)

--- a/addons/hr_recruitment/models/hr_recruitment_source.py
+++ b/addons/hr_recruitment/models/hr_recruitment_source.py
@@ -4,8 +4,7 @@
 from odoo import fields, models
 
 
-class RecruitmentSource(models.Model):
-    _name = "hr.recruitment.source"
+class HrRecruitmentSource(models.Model):
     _description = "Source of Applicants"
     _inherit = ['utm.source.mixin']
 

--- a/addons/hr_recruitment/models/hr_recruitment_stage.py
+++ b/addons/hr_recruitment/models/hr_recruitment_stage.py
@@ -4,8 +4,7 @@
 from odoo import api, fields, models, _
 
 
-class RecruitmentStage(models.Model):
-    _name = "hr.recruitment.stage"
+class HrRecruitmentStage(models.Model):
     _description = "Recruitment Stages"
     _order = 'sequence'
 
@@ -38,7 +37,7 @@ class RecruitmentStage(models.Model):
             context = dict(self._context)
             context.pop('default_job_id')
             self = self.with_context(context)
-        return super(RecruitmentStage, self).default_get(fields)
+        return super().default_get(fields)
 
     @api.depends('hired_stage')
     def _compute_is_warning_visible(self):

--- a/addons/hr_recruitment/models/ir_ui_menu.py
+++ b/addons/hr_recruitment/models/ir_ui_menu.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class IrUiMenu(models.Model):
-    _inherit = 'ir.ui.menu'
+    _inherit = ['ir.ui.menu']
 
     def _load_menus_blacklist(self):
         res = super()._load_menus_blacklist()

--- a/addons/hr_recruitment/models/mail_activity_plan.py
+++ b/addons/hr_recruitment/models/mail_activity_plan.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class MailActivityPlan(models.Model):
-    _inherit = 'mail.activity.plan'
+    _inherit = ['mail.activity.plan']
 
     def _compute_department_assignable(self):
         super()._compute_department_assignable()

--- a/addons/hr_recruitment/models/res_company.py
+++ b/addons/hr_recruitment/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = "res.company"
+    _inherit = ["res.company"]
 
     candidate_properties_definition = fields.PropertiesDefinition('Candidate Properties')
     job_properties_definition = fields.PropertiesDefinition("Job Properties")

--- a/addons/hr_recruitment/models/res_users.py
+++ b/addons/hr_recruitment/models/res_users.py
@@ -3,8 +3,9 @@
 
 from odoo import models
 
+
 class ResUsers(models.Model):
-    _inherit = 'res.users'
+    _inherit = ['res.users']
 
     def _create_recruitment_interviewers(self):
         if not self:

--- a/addons/hr_recruitment/models/utm_campaign.py
+++ b/addons/hr_recruitment/models/utm_campaign.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class UtmCampaign(models.Model):
-    _inherit = 'utm.campaign'
+    _inherit = ['utm.campaign']
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_utm_campaign_job(self):

--- a/addons/hr_recruitment/models/utm_source.py
+++ b/addons/hr_recruitment/models/utm_source.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class UtmSource(models.Model):
-    _inherit = 'utm.source'
+    _inherit = ['utm.source']
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_recruitment_sources(self):
